src/lib/uri/uri.c

```diff
@@ -213,3 +220,54 @@ uri_query_param_count(const struct uri *uri, const char *name)
 		uri_find_query_param(uri, name);
 	return (param != NULL ? param->value_count : 0);
 }
+
+void
+uri_set_add(struct uri_set *uri_set, struct uri *uri)
+{
+	size_t size = (uri_set->uri_count + 1) * sizeof(struct uri);
+	uri_set->uris = xrealloc(uri_set->uris, size);
+	uri_move(&uri_set->uris[uri_set->uri_count++], uri);
+}
+
+void
+uri_set_destroy(struct uri_set *uri_set)
+{
+	for (int i = 0; i < uri_set->uri_count; i++)
+		uri_destroy(&uri_set->uris[i]);
+	free(uri_set->uris);
+	TRASH(uri_set);
+}
+
+int
+uri_set_create(struct uri_set *uri_set, const char *str)
+{
+	uri_set->uris = NULL;
+	uri_set->uri_count = 0;
+	if (str == NULL || strlen(str) == 0)
+		return 0;
+	char *copy = xstrdup(str);
+	char *ptr = copy;
+	char *delim = NULL;
+	do {
+		char *uristr = ptr;
+		delim = strchr(ptr, ',');
+		if (delim != NULL) {
+			ptr = delim + 1;
+			*delim = '\0';
```

Comment:
No, strtok_r skips all consecutive charactersю For example "URI1,,URI2" in case when we use strtok splits in TWO correct
URI1 and URI2 uris. And in my implementation it splits in THREE uris: URI1, "", URI2. Empty uri is invalid so function fails.